File: app/src/main/java/com/sattar/j/smsfake/tools/customViews/CustomEditText.java
package com.sattar.j.smsfake.tools.customViews;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.Typeface;

import androidx.annotation.Nullable;

import android.util.AttributeSet;

import com.google.android.material.textfield.TextInputEditText;
import com.sattar.j.smsfake.R;
import com.sattar.j.smsfake.SmsFakeApplication;

public class CustomEditText extends TextInputEditText {
    private TypedArray textViewCustomFont;
    private int styleFont;
    public CustomEditText(Context context) {
        super(context);
        initFont(null);
    }

    public CustomEditText(Context context, AttributeSet attrs) {
        super(context, attrs);
        initFont(attrs);
    }

    private void initFont(AttributeSet attrs) {
        textViewCustomFont = getContext().obtainStyledAttributes(attrs, R.styleable.CustomFont);
        styleFont = textViewCustomFont.getInt(R.styleable.CustomFont_styleFont, 0);
        Typeface typeface = Typeface.createFromAsset(getContext().getAssets()
                , SmsFakeApplication.getFont(styleFont));
        setTypeface(typeface);
    }
}

File: app/src/main/java/com/sattar/j/smsfake/tools/customViews/CustomEditText.kt
package com.sattar.j.smsfake.tools.customViews

import android.content.Context
import android.content.res.TypedArray
import android.graphics.Typeface
import android.util.AttributeSet
import com.google.android.material.textfield.TextInputEditText
import com.sattar.j.smsfake.R
import com.sattar.j.smsfake.SmsFakeApplication.Companion.getFont

class CustomEditText : TextInputEditText {
    private var textViewCustomFont: TypedArray? = null
    private var styleFont = 0

    constructor(context: Context?) : super(context!!) {
        initFont(null)
    }

    constructor(context: Context?, attrs: AttributeSet?) : super(context!!, attrs) {
        initFont(attrs)
    }

    private fun initFont(attrs: AttributeSet?) {
        textViewCustomFont = context.obtainStyledAttributes(attrs, R.styleable.CustomFont)
        styleFont = textViewCustomFont!!.getInt(R.styleable.CustomFont_styleFont, 0)
        val typeface = Typeface.createFromAsset(context.assets, getFont(styleFont))
        setTypeface(typeface)
    }

    fun showBalloon(message:String){

    }
}